src/xenia/kernel/xam/xam_user.cc

```diff
@@ -684,14 +684,20 @@ dword_result_t XamSessionRefObjByHandle_entry(dword_t handle,
 }
 DECLARE_XAM_EXPORT1(XamSessionRefObjByHandle, kUserProfiles, kStub);
 
-dword_result_t XamUserIsUnsafeProgrammingAllowed_entry(
-    dword_t unk1, dword_t unk2, lpdword_t unk3, dword_t unk4, dword_t unk5,
-    dword_t unk6) {
-  if (!unk3 || unk1 != 255 && unk1 >= 4) {
-    return 87;
+dword_result_t XamUserIsUnsafeProgrammingAllowed_entry(dword_t user_index,
+                                                       dword_t unk,
+                                                       lpdword_t result_ptr) {
+  if (!result_ptr ||
```

Comment:
I would split this check into separate entry for lack of result pointer and incorrect user_index. That way it will be less confusing to read